tracer/src/Datadog.Trace/Tagging/AwsSnsTags.cs

```diff
@@ -3,19 +3,86 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
 // </copyright>
 
+using System;
 using Datadog.Trace.SourceGenerators;
 
+#pragma warning disable SA1402 // File must contain single type
 namespace Datadog.Trace.Tagging
 {
     internal partial class AwsSnsTags : AwsSdkTags
     {
+        public AwsSnsTags()
+            : this(SpanKinds.Client)
+        {
+        }
+
+        public AwsSnsTags(string spanKind)
+        {
+            SpanKind = spanKind;
+        }
+
         [Tag(Trace.Tags.AwsTopicName)]
         public string TopicName { get; set; }
 
         [Tag(Trace.Tags.AwsTopicArn)]
         public string TopicArn { get; set; }
 
         [Tag(Trace.Tags.SpanKind)]
-        public override string SpanKind => SpanKinds.Client;
+        public override string SpanKind { get; }
+    }
+
+    internal partial class AwsSnsV1Tags : AwsSnsTags
+    {
+        private string _peerServiceOverride = null;
+
+        // For the sake of unit tests, define a default constructor
+        // though the AWS SQS integration should use the constructor that takes a spanKind
```

Comment:
Some copy-pasta stuff

```suggestion
        // though the AWS SNS integration should use the constructor that takes a spanKind
```

Reply:
Happens to the best of us! 🤭